modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/factories/ResponseMessageProcessorsFactoryBean.java

```diff
@@ -46,20 +53,32 @@ public Object getObject() throws Exception {
       }
     }
     ResponseMessageProcessorAdapter responseAdapter = new ResponseMessageProcessorAdapter();
-    responseAdapter.setProcessor(builder.build());
+    responseAdapter.setProcessor(new Processor() {
+
+      private Processor processor;
+
+      @Override
+      public InternalEvent process(InternalEvent event) throws MuleException {
```

Comment:
Need to implement apply() also to delegate non-blocking invocation if this is required.